web/views.py

```
@@ -4715,88 +4717,66 @@ def create_donation_subscription(request):
-            customer_id = request.user.profile.stripe_customer_id
-
         # Create or get customer
-        if customer_id:
-            customer = stripe.Customer.retrieve(customer_id)
-        else:
+        customer = None
+        if request.user.is_authenticated and hasattr(request.user, "stripe_customer_id"):
+            try:
+                customer = stripe.Customer.retrieve(request.user.stripe_customer_id)
+            except stripe.error.InvalidRequestError:
+                pass
```

Ruff (0.8.2)

Use `contextlib.suppress(stripe.error.InvalidRequestError)` instead of `try`-`except`-`pass`

Replace with `contextlib.suppress(stripe.error.InvalidRequestError)`

(SIM105)